detector/cis/generic_linux/dockersocket/dockersocket.go

```diff
@@ -92,18 +93,33 @@ func (Detector) findingForTarget(target *inventory.GenericFindingTargetDetails)
 func (d Detector) ScanFS(ctx context.Context, fsys fs.FS, px *packageindex.PackageIndex) (inventory.Finding, error) {
 	var issues []string
 
+	// Check for context timeout
+	if ctx.Err() != nil {
+		return inventory.Finding{}, ctx.Err()
+	}
+
 	// Check Docker socket file permissions
-	if socketIssues := d.checkDockerSocketPermissions(fsys); len(socketIssues) > 0 {
+	if socketIssues := d.checkDockerSocketPermissions(ctx, fsys); len(socketIssues) > 0 {
```

Comment:
Hm, we're still testing the private functions in the first few tests.

It'd be good if we could use d.ScanFS in these individual tests as well - that way we can avoid testing private functions which is generally discouraged.